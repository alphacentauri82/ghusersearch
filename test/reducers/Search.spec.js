import searchReducer from 'reducers/Search';

describe('Reducer: search', () => {

  describe('when passed no state', () => {
    it('should return the initial state', () => {
      expect(
        searchReducer(undefined, {})
      ).toMatchSnapshot();
    });
  });

  describe('when a search request is pending', () => {
    it('should set isPending to true', () => {
      const beforeState = {
        isPending: false,
      };
      const afterState = searchReducer(beforeState, {type: 'SEARCH_REQUEST'});

      expect(
        afterState
      ).toMatchSnapshot();
    });
  });

  describe('when a search request is successful', () => {
    it('should update the state with the userIds', () => {
      const beforeState = {
        isPending: true,
        result: [],
      };
      const action = {
        payload: {
          result: [1, 2],
          search: '?q=alecrust',
        },
        type: 'SEARCH_SUCCESS',
      };
      const afterState = searchReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

  describe('when a search request fails', () => {
    it('should update the state with the error message', () => {
      const beforeState = {
        error: null,
        isPending: true,
      };
      const err = new Error('the server died');
      err.response = {foo: 'bar'};

      const action = {
        error: true,
        payload: err,
        type: 'SEARCH_FAILURE',
      };
      const afterState = searchReducer(beforeState, action);

      expect(
        afterState
      ).toMatchSnapshot();
      expect(afterState).not.toBe(beforeState);
    });
  });

});
