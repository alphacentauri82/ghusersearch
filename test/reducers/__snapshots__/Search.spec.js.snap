// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Reducer: search when a search request fails should update the state with the error message 1`] = `
Object {
  "error": [Error: the server died],
  "isPending": false,
}
`;

exports[`Reducer: search when a search request is pending should set isPending to true 1`] = `
Object {
  "isPending": true,
}
`;

exports[`Reducer: search when a search request is successful should update the state with the userIds 1`] = `
Object {
  "error": null,
  "isPending": false,
  "result": Array [
    1,
    2,
  ],
  "search": "?q=alecrust",
}
`;

exports[`Reducer: search when passed no state should return the initial state 1`] = `
Object {
  "error": null,
  "isPending": true,
  "pagination": null,
  "result": Array [],
  "totalResults": 0,
}
`;
